account_invoice_merge/invoice.py

```diff
@@ -188,9 +188,9 @@ def make_key(br, fields):
             if 'purchase.order' in self.env.registry else False
         for new_invoice_id in invoices_info:
             if so_obj:
-                todo_ids = so_obj.search(
+                todos = so_obj.search(
                     [('invoice_ids', 'in', invoices_info[new_invoice_id])])
-                for org_so in so_obj.browse(todo_ids):
+                for org_so in todos:
                     org_so.write({'invoice_ids': [(4, new_invoice_id)]})
```

Comment:
Following @pedrobaeza comment, you can extract write for loop and do

    todos.write({'invoice_ids': [(4, new_invoice_id)]})